pkg/validate/apparmor.go

```diff
@@ -0,0 +1,193 @@
+				checkContainerApparmor(rc, containerID, true)
+			})
+		})
+	}
+})
+
+func createContainerWithAppArmor(rc internalapi.RuntimeService, ic internalapi.ImageManagerService, sandboxID string, sandboxConfig *runtimeapi.PodSandboxConfig, profile string, shouldStart bool) string {
+	By("create a container with apparmor")
+	containerName := "apparmor-test-" + framework.NewUUID()
+	containerConfig := &runtimeapi.ContainerConfig{
+		Metadata: framework.BuildContainerMetadata(containerName, framework.DefaultAttempt),
+		Image:    &runtimeapi.ImageSpec{Image: framework.DefaultContainerImage},
+		Command:  []string{"touch", "foo"},
+		Linux: &runtimeapi.LinuxContainerConfig{
+			SecurityContext: &runtimeapi.LinuxContainerSecurityContext{
+				ApparmorProfile: profile,
+			},
+		},
+	}
+	containerID := framework.CreateContainer(rc, ic, containerConfig, sandboxID, sandboxConfig)
+
+	By("start container with apparmor")
+	err := rc.StartContainer(containerID)
+	if shouldStart {
+		Expect(err).NotTo(HaveOccurred())
+	} else {
+		Expect(err).To(HaveOccurred())
+	}
+
+	// wait container started and check the status.
+	Eventually(func() runtimeapi.ContainerState {
+		return getContainerStatus(rc, containerID).State
+	}, time.Minute, time.Second*4).Should(Equal(runtimeapi.ContainerState_CONTAINER_EXITED))
+
+	return containerID
+}
+
+func checkContainerApparmor(rc internalapi.RuntimeService, containerID string, shoudRun bool) {
+	By("get container status")
+	status, err := rc.ContainerStatus(containerID)
+	Expect(err).NotTo(HaveOccurred())
+
+	if shoudRun {
+		Expect(status.GetExitCode()).To(Equal(int32(0)))
+	} else {
+		Expect(status.GetExitCode()).NotTo(Equal(int32(0)))
+	}
+}
+
+func loadTestProfiles() error {
+	f, err := ioutil.TempFile("/tmp", "apparmor")
+	if err != nil {
+		return fmt.Errorf("failed to open temp file: %v", err)
+	}
+	defer os.Remove(f.Name())
+	defer f.Close()
+
+	// write test profiles to a temp file.
+	if _, err = f.WriteString(testProfiles); err != nil {
+		return fmt.Errorf("failed to write profiles to file: %v", err)
+	}
+
+	// load apparmor profiles into kernel.
+	cmd := exec.Command("sudo", "apparmor_parser", "-r", "-W", f.Name())
+	stderr := &bytes.Buffer{}
+	cmd.Stderr = stderr
+	out, err := cmd.Output()
+	// apparmor_parser does not always return an error code, so consider any stderr output an error.
+	if err != nil || stderr.Len() > 0 {
+		if stderr.Len() > 0 {
+			glog.Warning(stderr.String())
+		}
+		if len(out) > 0 {
+			glog.Infof("apparmor_parser: %s", out)
+		}
+
+		return fmt.Errorf("failed to load profiles: %v", err)
+	}
+
+	glog.V(2).Infof("Loaded profiles: %v", out)
+	return nil
+}
+
+// isAppArmorEnabled returns true if apparmor is enabled for the host.
+// This function is forked from
+// https://github.com/opencontainers/runc/blob/1a81e9ab1f138c091fe5c86d0883f87716088527/libcontainer/apparmor/apparmor.go
+// to avoid the libapparmor dependency.
```

Comment:
Add TODO to reuse the function after we could properly sync api?

Reply:
ack